### DISCUSSION TITLE
Conductor CLI

### DISCUSSION THREAD

**Discussion by NickTomlin** [Netflix/conductor#2560] [Ideas]
I've been using a node.js based `cli` to simplify my conductor workflow and wanted to share it with you all to get some feedback. Here's a quick video overview:

<a href="https://www.loom.com/share/970a237d3d884bf88bad7492111079fa">
    <p>Conductor CLI demo - Watch Video</p>
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/970a237d3d884bf88bad7492111079fa-with-play.gif">
  </a>

# Feature overview
(if you don't want to wait for the video, here's a brief text overview)

## Running workflows

This is the feature I personally use the most, it's helpful to have a quick feedback loop when running workflow definitions.

```
##### Remote workflows #####
## execute a remote workflow with `workflowDef`
$ conductor workflow:run my.remote.workflow --overrideJson '{ 
    "input": { "name": "Nick" }
}'
66b15471-ea1a-4d43-8f94-3bd32dbd5c31 # UUID of execution

## execute a remote workflow with `workflowDef` provided by a file
$ cat simple/simple.input.json
{
  "input": {
    "names": ["Eleven", "Hopper", "Max", "Dustin"]
  }
}
$ conductor workflow:run my.remote.workflow --overrideFile simple/simple.input.json
66b15471-ea1a-4d43-8f94-3bd32dbd5c31 # UUID of execution


##### Local workflows #####
# a workflow def that is on my local machine
# (all the input options above apply)
$ conductor workflow:run simple/simple.conductor.json
66b15471-ea1a-4d43-8f94-3bd32dbd5c31 # UUID of execution
```

## Opening in the `conductor` UI

The `workflow:run` command takes an `--open` flag that opens the workflow execution in the conductor UI.

## Updating workflow and task metadata

```shell
##### Update workflows #####
$ conductor workflow:update simple/simple.conductor.json complex/complex.conductor.json

##### Update workflows #####
$ conductor tasks:update tasks/my-task.json
```

## Use cases

- Easier onboarding / getting started
- Simplify and speed up development

#  ❓ Questions ❓ 

I'm interested in opinions here, is this worth open sourcing? Are there different directions we should head in?

**Comment by sso679** [comment]
when i do changes in this .json files in conductor-resources /workflow.recorder.initiated.regestration.json and triggered the pipeline and when i see in conductor ui ,i dont see the changes affected in the UI but when i do any changes in .ts file ,conductor ui is picking up changes done in .json files.
could anyone help me please.

**Reply by NickTomlin** [reply]
Hi there @sso679, if you are still experiencing an issue can you start a new thread to focus on that? Thanks!
